cmd/client.go

```diff
@@ -65,15 +66,16 @@ func NewClientFactory(n func() *fn.Client) ClientFactory {
 //	defer done()
 func NewClient(cfg ClientConfig, options ...fn.Option) (*fn.Client, func()) {
 	var (
-		p  = progress.New(cfg.Verbose)            // updates the CLI
-		t  = newTransport(cfg.InsecureSkipVerify) // may provide a custom impl which proxies
-		c  = newCredentialsProvider(t)            // for accessing registries
+		p  = progress.New(cfg.Verbose)                // updates the CLI
+		t  = newTransport(cfg.InsecureSkipVerify)     // may provide a custom impl which proxies
+		c  = newCredentialsProvider(config.Path(), t) // for accessing registries
```

Comment:
Passing in the path both removes the implicit read of an environment variable from non-main, and avoids a circular dependency with config.